File: Python/2024/2407/240702/BAEK_15656.py
def recur(cnt, output):
    if cnt == M:
        result.append(' '.join(map(str, output)))
        return
    
    for e in arr:
        output.append(e)
        recur(cnt+1, output)
        output.pop()


N, M = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()
result = []
recur(0, [])
print('\n'.join(result))


# def recur(cnt, output):
#     if cnt == M:
#         print(*output)
#         return
    
#     for e in arr:
#         output.append(e)
#         recur(cnt+1, output)
#         output.pop()


# N, M = map(int, input().split())
# arr = list(map(int, input().split()))
# arr.sort()
# recur(0, [])

File: JavaScript/2407/240702/BAEK_15656.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const recur = (cnt, output) => {
  if (cnt === M) {
    result.push(output.join(' '));
    return;
  }
  for (const e of arr) {
    output.push(e);
    recur(cnt + 1, output);
    output.pop();
  }
}

let N, M;
let arr;
const result = [];
let idx = 0;

rl.on('line', (line) => {
  if (idx === 0) {
    [N, M] = line.split(' ').map(Number);
  } else {
    arr = Array.from(line.split(' ').map(Number));
  }
  idx++;
  if (idx > 1) {
    rl.close();
  }
}).on('close', () => {
  arr.sort((a, b) => a - b);
  recur(0, []);
  console.log(result.join('\n'));
});